tests/unit/lib/Gocdb_Services/ServiceServiceTest.php

```diff
@@ -0,0 +1,195 @@
+    //return PHPUnit_Extensions_Database_Operation_Factory::NONE();
+    //
+    // Issue a DELETE from <table> which is more portable than a
+    // TRUNCATE table <table> (some DBs require high privileges for truncate statements
+    // and also do not allow truncates across tables with FK contstraints e.g. Oracle)
+    return PHPUnit_Extensions_Database_Operation_Factory::DELETE_ALL();
+  }
+
+  /**
+  * Overridden.
+  */
+  protected function getTearDownOperation() {
+    // NONE is default
+    return PHPUnit_Extensions_Database_Operation_Factory::NONE();
+  }
+
+  /**
+  * Sets up the fixture, e.g create a new entityManager for each test run
+  * This method is called before each test method is executed.
+  */
+  protected function setUp() {
+    parent::setUp();
+    $this->em = $this->createEntityManager();
+  }
+
+  /**
+  * @todo Still need to setup connection to different databases.
+  * @return EntityManager
+  */
+  private function createEntityManager(){
+    require __DIR__ . '/../../../doctrine/bootstrap_doctrine.php';
+    return $entityManager;
+  }
+
+  /**
+  * Called after setUp() and before each test. Used for common assertions
+  * across all tests.
+  */
+  protected function assertPreConditions() {
+    $con = $this->getConnection();
+    $fixture = __DIR__ . '/../../../doctrine/truncateDataTables.xml';
+    $tables = simplexml_load_file($fixture);
+
+    foreach($tables as $tableName) {
+      $sql = "SELECT * FROM ".$tableName->getName();
+      $result = $con->createQueryTable('results_table', $sql);
+      if($result->getRowCount() != 0){
+        throw new RuntimeException("Invalid fixture. Table has rows: ".$tableName->getName());
+      }
+    }
+  }
+
+  public function createValidationEntities(){
+    
+    $site = TestUtil::createSampleSite("TestSite");
+    $this->em->persist($site);
+
+    $service = TestUtil::createSampleService("TestService1");
+    $this->em->persist($service);
+    $service->setParentSiteDoJoin($site);
+
+    $user = TestUtil::createSampleUser("Test", "Testing", "/c=test");
+    $this->em->persist($user);
+    $user->setAdmin(TRUE);
+
+    $roleActionMappingService = new org\gocdb\services\RoleActionMappingService();
+    $roleActionAuthService = new org\gocdb\services\RoleActionAuthorisationService($roleActionMappingService);
+    $roleActionAuthService->setEntityManager($this->em);
+
+    $this->em->flush();
+
+    $serviceService = new org\gocdb\services\ServiceService();
+    $serviceService->setEntityManager($this->em);
+    $serviceService->setRoleActionAuthorisationService($roleActionAuthService);
+
+    return array($service, $user, $serviceService);
+  }
+  /**
+   * Check the basics -
+   * Duplicates some simple testing from ExtensionsTest
+   */
+  public function testValidateProperty(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("ValidName1","ValidValue");
```

Comment:
are these arrays of the form ("Property Name", Property Value")?

Reply:
Yes.